### PR TITLE
Fail2ban ssh protection integration in Ansible playbooks.

### PR DESCRIPTION
**Pull request by valkov-ivaylo**
Closes https://www.pivotaltracker.com/n/projects/2152594/stories/160687547 

Tested in [CircleCI](https://circleci.com/gh/aeternity/infrastructure/4276) and manually on localhost.

### CODE REVIEW

File: ansible/setup.yml [hunk 1]
```diff
@@ -24,6 +24,16 @@
       - sylane
     libsodium_version: "1.0.16"
     libsodium_module_version: "23.1.0"
+    fail2ban_loglevel: INFO
+    fail2ban_bantime: 86400
+    fail2ban_maxretry: 3
```
Comment (lucafavatella) [inline ansible/setup.yml]: Have you tested whether this 3 affects typical error of user attempting to SSH forgetting to load SSH key in agent in the first place? I am not sure if SSH CLI attempts multiple times under the hood, and how this behaves for a couple of users in same building.

(Distinct case of user's keypair change is expected to have a few bumps for at least a day.)

File: ansible/setup.yml [hunk 2]
```diff
@@ -24,6 +24,16 @@
...
+    fail2ban_logtarget: SYSLOG
```
[the remark this answers is not in the document]
Comment (valkov-ivaylo) [reply ansible/setup.yml]: According to the defualt fail2ban.conf (/etc/fail2ban/fail2ban.conf) the supported log levels are:

```
# Option: loglevel
# Notes.: Set the log level output.
#         CRITICAL
#         ERROR
#         WARNING
#         NOTICE
#         INFO
#         DEBUG

```
It should be picked by the datadog agent and the [rsyslog config for datadog](https://github.com/aeternity/infrastructure/blob/master/ansible/files/rsyslog/datadog.conf).

File: ansible/monitoring.yml [hunk 3]
```diff
@@ -135,6 +135,32 @@
...
+      notify: restart rsyslogd
+    - name: Copy fail2ban.conf to logrotate
+      copy:
+        src: files/logrotate/fail2ban
+        dest: /etc/logrotate.d/fail2ban
+        mode: 0600
+    - name: Copy fail2ban.log with correct permissions for syslog
+      copy:
+        content: ""
+        dest: /var/log/fail2ban.log
```
[the remark this answers is not in the document]
Comment (valkov-ivaylo) [reply ansible/monitoring.yml]: This was intended only as a fix for permissions required by the rsyslog daemon and the SYSLOG logtarget for fail2ban. I don't think it is correct to be reset every time. Any suggestions to fix this?

[the remark this answers is not in the document]
Comment (lucafavatella) [reply ansible/monitoring.yml]: It seems you wanted:
```
- file:
    path: /var/log/fail2ban.log
    state: touch
    mode: 0640
    owner: syslog
    group: adm
    modification_time: "preserve" ## Only ansible 2.7 or greater.
    access_time: "preserve" ## Only ansible 2.7 or greater.
```

(From https://docs.ansible.com/ansible/latest/modules/file_module.html )

I did not test this, I am not sure what is impact of resetting file to empty, hence not sure if worth the effort.

File: ansible/setup.yml (hunk 1 above)
Comment (valkov-ivaylo) [reply ansible/setup.yml]: Every invalid access attempt (password or key) will be detected. So using a wrong key and switching to the correct one, will be treated as one failed attempt. Multiple users behind the same IP address will all have 3 attempts for the same IP address. 

Fail2ban has a configuration option for the time-frame (findtime) for those attempts, but it is not configurable with the current Ansible role. The default (in the distribution provided global config) is 600 seconds. We could increase the retries to work-around those issues.

**Comment by valkov-ivaylo** [comment]
Requires #141

**Pull request merged**